samza-core/src/main/java/org/apache/samza/operators/impl/AsyncStreamOperatorImpl.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.samza.operators.impl;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.concurrent.CompletionStage;
+import java.util.concurrent.ExecutionException;
+import org.apache.samza.SamzaException;
+import org.apache.samza.context.Context;
+import org.apache.samza.operators.functions.AsyncFlatMapFunction;
+import org.apache.samza.operators.spec.AsyncOperatorSpec;
+import org.apache.samza.operators.spec.OperatorSpec;
+import org.apache.samza.task.MessageCollector;
+import org.apache.samza.task.TaskCoordinator;
+
+
+public class AsyncStreamOperatorImpl<M, RM> extends OperatorImpl<M, RM> {
+  private final AsyncOperatorSpec<M, RM> asyncOperatorSpec;
+  private final AsyncFlatMapFunction<M, RM> transformFn;
+
+  AsyncStreamOperatorImpl(AsyncOperatorSpec<M, RM> asyncOperatorSpec) {
+    this.asyncOperatorSpec = asyncOperatorSpec;
+    this.transformFn = asyncOperatorSpec.getTransformFn();
+  }
+  @Override
+  protected void handleInit(Context context) {
+    this.transformFn.init(context);
+  }
+
+  @Override
+  protected Collection<RM> handleMessage(M message, MessageCollector collector, TaskCoordinator coordinator) {
```

Comment:
Fixed it. Although I left the `onMessage` and `handleMessage` as package private test helper methods otherwise, tests would need to convert the `CompletionStage` to `CompletableFuture` and do a get on them. 

I can follow up with another PR just targeting the tests cleanup if we feel tests should be okay doing the above and we don't need a package private helper method.